src/pip/_internal/req/req_uninstall.py

```diff
@@ -74,8 +74,27 @@ def uninstallation_paths(dist):
     the .pyc and .pyo in the same directory.
 
     UninstallPathSet.add() takes care of the __pycache__ .py[co].
+
+    If RECORD is not found, raises UninstallationError,
+    with possible information from the INSTALLER file.
+
+    https://packaging.python.org/specifications/recording-installed-packages/
     """
-    r = csv.reader(dist.get_metadata_lines('RECORD'))
+    try:
+        r = csv.reader(dist.get_metadata_lines('RECORD'))
+    except FileNotFoundError as missing_record_exception:
+        msg = 'Cannot uninstall {dist}, RECORD file not found.'.format(dist=dist)
+        try:
+            installer = next(dist.get_metadata_lines('INSTALLER'))
+            if not installer or installer == 'pip':
+                raise ValueError()
+        except (OSError, StopIteration, ValueError):
+            dep = '{}=={}'.format(dist.project_name, dist.version)
+            msg += (' You might be able to recover from this via: '
+                    '`pip install --force-reinstall --no-deps {}`.'.format(dep))
```

Comment:
I think pip generally uses straight quotes in the message instead (although inconsistent whether to use single or double quotes). I think I prefer single quotes here.

Reply:
Done.